hawkular-inventory-bus/src/main/java/org/hawkular/inventory/bus/MessageSender.java

```diff
@@ -35,24 +36,25 @@
  * @since 0.0.1
  */
 final class MessageSender {
-    private final MessageProcessor messageProcessor;
-    private final ConnectionContextFactory connectionContextFactory;
     private final String topicName;
-    private ProducerConnectionContext producerConnectionContext;
+    private final TopicConnectionFactory topicConnectionFactory;
+    private final MessageProcessor messageProcessor;
 
+    public MessageSender(TopicConnectionFactory  topicConnectionFactory, String topicName) {
```

Comment:
I agree with @pavolloffay. We can change this if and when the need for sending to a queue arises.